lang_GENERIC/analyze/Naming_AST.ml

```diff
@@ -415,7 +439,8 @@ let resolve lang prog =
         VarDef ({ vinit; vtype }) when is_local_or_global_ctx env ->
           (* name resolution *)
           let sid = Ast.gensym () in
-          let resolved = { entname = resolved_name_kind env, sid; enttype = vtype } in
+          let resolved_type = get_resolved_type (vinit, vtype) in
```

Comment:
maybe a comment here saying we either use the optional type in vtype, or if we can infer the type
of the expression vinit if it's a simple literal we use that as a type.

Reply:
maybe can also say "this is especially useful in Go where you can do var x = 2 without having to declare the type"